pkg/trace/obfuscate/json.go

```diff
@@ -25,27 +27,47 @@ func (o *Obfuscator) obfuscateJSON(span *pb.Span, tag string, obfuscator *jsonOb
 	// as much of it as we could. It is safe to accept the output, even if partial.
 }
 
+type valueTransformer func(string) string
+
 type jsonObfuscator struct {
-	keepers map[string]bool // these keys will not be obfuscated
+	keepKeys      map[string]bool // these keys will not be obfuscated
+	transformKeys map[string]bool // these keys pass through the transformer
+	transformer   valueTransformer
 
 	scan     *scanner // scanner
 	closures []bool   // closure stack, true if object (e.g. {[{ => []bool{true, false, true})
 	key      bool     // true if scanning a key
 
-	wiped     bool // true if obfuscation string (`"?"`) was already written for current value
-	keeping   bool // true if not obfuscating
-	keepDepth int  // the depth at which we've stopped obfuscating
+	wiped             bool // true if obfuscation string (`"?"`) was already written for current value
+	keeping           bool // true if not obfuscating
+	transformingValue bool // true if collecting the next literal for transformation
+	keepDepth         int  // the depth at which we've stopped obfuscating
 }
 
-func newJSONObfuscator(cfg *config.JSONObfuscationConfig) *jsonObfuscator {
+func (o *Obfuscator) newJSONObfuscator(cfg *config.JSONObfuscationConfig) *jsonObfuscator {
 	keepValue := make(map[string]bool, len(cfg.KeepValues))
 	for _, v := range cfg.KeepValues {
 		keepValue[v] = true
 	}
+	var transformValues map[string]bool
+	var transformer valueTransformer
```

Comment:
I'd propose
```suggestion
	var (
	    transformVals map[string]bool
	    transformFunc valueTransformer
	)
```

Reply:
Sure. It takes double the number of lines though 😛